SampleApp/Components/ShadowView.swift

```diff
@@ -43,6 +42,23 @@ struct ShadowView: View {
     }
 }
 
+private struct ShadowButtonStyle: ButtonStyle {
```

Comment:
I don't want to make this specifically a re-usable thing because users might want more than just an interactive shadow in their custom button style, or maybe something else than a button style altogether. The idea of this PR is simply to make a pressed state available without being too opinionated on how users will use it.

With that said, it probably wouldn't be a bad idea to make something like a `CardViewButtonStyle` that not only standardises the shadow pressed state, but the look of card view buttons in general. That is at least what I'm planning on using this for in Jobs. I see that as something for another PR though.